azext_iot/operations/hub.py

```diff
@@ -2041,14 +2042,45 @@ def _handle_c2d_msg(target, device_id, receive_settle):
 
 
 def iot_device_export(
-    cmd, hub_name, blob_container_uri, include_keys=False, resource_group_name=None
+    cmd,
+    hub_name,
+    blob_container_uri,
+    include_keys=False,
+    storage_authentication_type=None,
+    resource_group_name=None,
 ):
     from azext_iot._factory import iot_hub_service_factory
+    from azure.mgmt.iothub import __version__ as iot_sdk_version
 
     client = iot_hub_service_factory(cmd.cli_ctx)
     target = get_iot_hub_connection_string(client, hub_name, resource_group_name)
+
+    if ensure_min_version(iot_sdk_version, "0.12.0"):
+        from azure.mgmt.iothub.models import ExportDevicesRequest
+        from azext_iot.common.shared import AuthenticationType
+
+        storage_authentication_type = (
+            AuthenticationType(storage_authentication_type).name
+            if storage_authentication_type
+            else None
+        )
+        export_request = ExportDevicesRequest(
+            export_blob_container_uri=blob_container_uri,
+            exclude_keys=not include_keys,
```

Comment:
nit - this `not include_keys` is a little wierd, and I see that both usages of it in this function are using `exclude_keys`. Would it be worth while changing the function definition to be `exclude_keys` instead?

Reply:
Yeah, I agree it's a bit wonky. This command interface was pre-extension (way back when) and it's likely been kept the way it is to avoid a breaking change.

We can eventually deprecate `include_keys` as a parameter in place of `exclude_keys` to match the SDK model and parameters, like you mentioned, but I'd rather address that as a separate PR / issue.